native/packages/react-native-bpk-component-animate-height/src/BpkAnimateHeight.js

```diff
@@ -34,92 +33,114 @@ const {
   animationDurationSm,
 } = tokens;
 
+const collapsedHeight = 0.01;
+
 class BpkAnimateHeight extends React.Component {
-  constructor() {
-    super();
+  constructor(props) {
+    super(props);
 
     this.state = {
-      expanded: null,
-      expandedHeight: null,
-      collapsedHeight: 0,
-      height: new Animated.Value(1),
-      heightSet: false,
+      expanded: false,
+      height: new Animated.Value(0.01),
+      inTree: false,
+      expandAnimationInProgress: false,
     };
 
+    // overflow: hidden is not properly supported on Android,
+    // which causes animation to look shoddy.
+    // See https://facebook.github.io/react-native/releases/0.49/docs/layout-props.html#overflow
+    this.animationDuration = Platform.OS === 'ios' ? props.animationDuration : 0;
+
     this.componentWillReceiveProps = this.componentWillReceiveProps.bind(this);
     this.setExpandedHeight = this.setExpandedHeight.bind(this);
-    this.resize = this.resize.bind(this);
+    this.onHeightAnimationComplete = this.onHeightAnimationComplete.bind(this);
   }
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.expanded !== this.state.expanded) {
-      this.state.expanded = nextProps.expanded;
-      this.resize();
+      if (this.state.expanded) {
+        this.setState({ expanded: false });
+        Animated.timing(this.state.height, {
+          toValue: collapsedHeight,
+          duration: this.animationDuration,
+          delay: this.props.collapseDelay,
+        }).start(this.onHeightAnimationComplete);
+      } else {
+        this.setState({ inTree: true, expandAnimationInProgress: true });
+      }
     }
   }
 
-  setExpandedHeight(event, expandedImmediately) {
-    const { height } = event.nativeEvent.layout;
-    this.setState({
-      // if the component has started expanded, we should now set the height of the container
-      // why on earth does a value of 1 instead of 0 make this break!
-      height: new Animated.Value(expandedImmediately ? height : 0),
-      heightSet: true,
-      expandedHeight: height,
-    });
+  onHeightAnimationComplete() {
+    if (typeof this.props.onAnimationComplete === 'function') {
+      this.props.onAnimationComplete();
+    }
+    if (this.state.height._value - 0.01 < 0.00001) { // eslint-disable-line no-underscore-dangle
```

Comment:
Is this the best way to get the value?

Reply:
It does seem to be. An issue was raised requesting that `getValue()` be made public, but it was not deemed necessary